sw/host/ot_certs/src/template/mod.rs

```diff
@@ -0,0 +1,334 @@
+//! OpenTitan certificate template deserialization.
+//!
+//! This module contains structs defining certificate templates.
+//!
+//! These structs are defined in Hjson files and deserialized to here. Any
+//! extra conversion required (beyond simple renaming) is done in the `hjson`
+//! module.
+//!
+//! The format for a template in Hjson looks something like:
+//!
+//! ```hjson
+//! {
+//!   variables: {
+//!     SomeVariableName: {
+//!       type: "byte-array",
+//!       size: 20,
+//!     },
+//!     // ...
+//!   },
+//!
+//!   certificate: {
+//!     // Certificate keys, some making use of variables, others not.
+//!     serial_number: { var: "SomeVariableName" },
+//!     layer: 0,
+//!     // ...
+//!   }
+//! }
+//! ```
+
+use anyhow::Result;
+use num_bigint_dig::BigUint;
+use serde::{Deserialize, Deserializer};
+use serde_with::{serde_as, As, DeserializeAs, Same};
+use std::collections::HashMap;
+
+mod hjson;
+
+/// Full template file, including variable declarations and certificate spec.
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
+pub struct Template {
+    /// Name of the certificate.
+    pub name: String,
+    /// Variable declarations.
+    // NOTE: we need to use `Same` since String does not implement `DeserializeAs`.
+    #[serde_as(as = "HashMap<Same, hjson::VariableType>")]
+    pub variables: HashMap<String, VariableType>,
+    /// Certificate specification.
+    pub certificate: Certificate,
+}
+
+/// Certificate specification.
+#[serde_as]
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize)]
+pub struct Certificate {
+    #[serde_as(as = "Value<hjson::BigUint>")]
+    pub serial_number: Value<BigUint>,
+    pub issuer: HashMap<AttributeType, Value<String>>,
+    pub subject: HashMap<AttributeType, Value<String>>,
+    pub subject_public_key_info: SubjectPublicKeyInfo,
+    #[serde_as(as = "Value<hjson::HexString>")]
+    pub authority_key_identifier: Value<Vec<u8>>,
+    #[serde_as(as = "Value<hjson::HexString>")]
```

Comment:
maybe add some comments here indicating which parts are standard parts of X.509 certs vs which are custom parts of the DiceTcbInfo extension?